hibernate-core/src/main/java/org/hibernate/jpa/boot/internal/EntityManagerFactoryBuilderImpl.java

```diff
@@ -192,20 +192,17 @@ private EntityManagerFactoryBuilderImpl(
 
 		this.persistenceUnit = persistenceUnit;
 
-		if ( integrationSettings == null ) {
-			integrationSettings = Collections.emptyMap();
-		}
+		final Map integtrationSettingsCopy = integrationSettings == null ? new HashMap() : new HashMap( integrationSettings );
 
-		Map mergedIntegrationSettings = null;
 		Properties properties = persistenceUnit.getProperties();
 		if ( properties != null ) {
-			mergedIntegrationSettings = new HashMap( persistenceUnit.getProperties() );
-			mergedIntegrationSettings.putAll( integrationSettings );
+			// original integration settings take precedence
+			properties.forEach( integtrationSettingsCopy::putIfAbsent );
 		}
 
 		// Build the boot-strap service registry, which mainly handles class loader interactions
 		final BootstrapServiceRegistry bsr = buildBootstrapServiceRegistry(
-				mergedIntegrationSettings != null ? mergedIntegrationSettings : integrationSettings,
+				integtrationSettingsCopy,
```

Comment:
The original implementation doesn't enjoy crystal clear logic for it is tightly coupled with previous logic. Centralizing the logic in one single location up front is better, IMO.